### DISCUSSION TITLE
Update a child record from the parent

### DISCUSSION THREAD

**Discussion by gvlx** [Budibase/budibase#3365] [Show and tell]
Following this guide https://docs.budibase.com/guides/create-a-child-record-from-the-parent, it is very easy to _create_ child records.

But to _edit_ them, what do you need?

In this example, I have the child table `child` related to the parent table  `main` through the field `rel_main` which has the reverse field in the parent table `rev_child`:
![imagem](https://user-images.githubusercontent.com/67311/141654369-83c065f5-82ac-4d64-b499-c6453b43cba0.png)

In the parent's screen (called with the parent's `id`) with a table of its children `table_child` (could also be a repeater) bound to the `rev_child` column.

![imagem](https://user-images.githubusercontent.com/67311/141654188-ddbd03cd-87dc-4cbd-91c3-a9d711608583.png)

![imagem](https://user-images.githubusercontent.com/67311/141654294-70c22f28-b142-4216-8ab5-b909c18d630b.png)

Add an icon (in this case `pencil-fill`) with an action `Navigate to` with the url `/child/{{ table_child.child._id}}`

![imagem](https://user-images.githubusercontent.com/67311/141654316-0eb80112-41d7-44b6-88be-1b1e08408809.png)

`{{ table_child.child._id}}` is the id of the child record select, as `rev_child` is the reverse field in the relation, it contains a collection of all its children, with their id and display value, which are individually exposed in each row of `table_child`.

This way you can call directly the child's edit screen make any changes and save it.

You can use the same approach of using the model screens as in the guide above, to avoid refreshing the screen too much.

**Comment by gvlx** [comment]
**Problem**

If _not using the modal approach_, the child's edit screen `Save` button should have an action `Navigate to` with the url `//main/with_child/{{ Repeater.child.rel_main }}`. 

**But this won't work** as that field contains the _display_name_ of the parent, not its id not even an object id+display name like in the reverse field above.

There is an easy solution, but it requires some 'extra magical steps'... 🪄

**Solution**
Basically, we need to create a new column on the child, formula, which captures the parent's id: 
![imagem](https://user-images.githubusercontent.com/67311/141655056-b22c3776-1694-42df-8f92-852a49fb3339.png)

The result:
![imagem](https://user-images.githubusercontent.com/67311/141655088-deb3931a-16e9-42c3-995f-057847b2f710.png)

This value is now available in the child's edit screen and can be used to navigate to the parent's record:
![imagem](https://user-images.githubusercontent.com/67311/141655156-4ff8ef19-e3dd-48c5-94df-67e7a658909d.png)

**Reply by gvlx** [reply]
**Where are the 'extra magical steps'?** 🪄

Well, you see, in the `Handlebars Formula` help screen of the formula column, there is never any indication of such `_id` field:

![imagem](https://user-images.githubusercontent.com/67311/141655249-52c87b9b-fef8-4962-9a5c-12da3f3fb002.png)

But, in same cases, it is shown (here is helper in the parent's screen):
![imagem](https://user-images.githubusercontent.com/67311/141655271-055e75a7-b4f9-4623-aa0b-ed58524f7736.png)

SO, I USED IT! 🤣 🪄

**Reply by mjashanks** [reply]
Nice write-up - thanks!

Another way you could achieve this, is by passing the Parent's `_id` into the edit screen, via the URL. So, you would change the edit screen to be:

`/child/:parent_id/:child_id`

Now, inside the Child screen, you have access to the parent `_id` via `{{ URL.parent_id }}`

**Reply by gvlx** [reply]
Nice, I didn't knew that one. :)
